browser_history/generic.py

```diff
@@ -101,11 +131,18 @@ def __init__(self, plat: utils.Platform = None):
             assert self.linux_path is not None, error_string.format("Linux")
             self.history_dir = homedir / self.linux_path
         else:
-            self.history_dir = None
+            raise NotImplementedError()
 
         if self.profile_support and not self.profile_dir_prefixes:
             self.profile_dir_prefixes.append("*")
 
+    def bookmarks_parser(
+        self, bookmark_path: str
+    ) -> typing.Optional[
+        typing.Tuple[datetime.datetime, str, str, str]
+    ]:  # pylint: disable=assignment-from-no-return
```

Comment:
Maybe add `return None` instead of disabling the warning?

Reply:
> Maybe add return None instead of disabling the warning?

it will raise another warning

```[assignment-from-none] Assigning result of a function call, where the function returns None [E1128]```

---

mypy check failed on Browser class on browser module

```Error: Incompatible types in assignment (expression has type "Callable[[Chrome, Any], Any]", base class "Browser" defined the type as "Callable[[Browser, str], Optional[Tuple[datetime, str, str, str]]]")```

i may have to disable type hint to pass this for now

Reply:
Okay.
Those two types should be equivalent, right?

Reply:
>> "Callable[[Browser, str], Optional[Tuple[datetime, str, str, str]]]"
>> "Callable[[Chrome, Any], Any]"

>  Those two types should be equivalent, right?

no. to match first type hint browser class have to inherit another custom browser class

so for example below will not match

```python 
class Brave(Browser):
    ...
    bookmarks_parser = Chrome.bookmarks_parser
```
it should be this instead

```python 
class Brave(Chrome):
    ...
```
this way Brave.bookmarks_parser will inherit Chrome.bookmarks_parser